soroban-env-host/src/host/conversion.rs

```diff
@@ -500,7 +501,13 @@ impl Host {
             }
             ScVal::Bytes(b) => Ok(self.add_host_object(b.metered_clone(self)?)?.into()),
             ScVal::String(s) => Ok(self.add_host_object(s.metered_clone(self)?)?.into()),
-            ScVal::Symbol(s) => Ok(self.add_host_object(s.metered_clone(self)?)?.into()),
```

Comment:
I'm curious - What path was this code previously taking? It's a little hard to follow.

Reply:
There is actually no additional code path here - `add_host_object` is infallible, so it used to just create a symbol object unconditionally.